api/lcp/server.py

```diff
@@ -0,0 +1,351 @@
+        ),
+        type=ConfigurationAttributeType.NUMBER,
+        required=False,
+        default=DEFAULT_MAX_COPIABLE_PAGES
+    )
+
+
+class LCPServer(object):
+    """Wrapper around LCP License Server's API"""
+
+    DEFAULT_PATRON_PASSPHRASE = 'patron'
+
+    def __init__(self, configuration_storage, configuration_factory, hasher_factory):
+        """Initializes a new instance of LCPServer class
+
+        :param configuration_storage: ConfigurationStorage object
+        :type configuration_storage: ConfigurationStorage
+
+        :param configuration_factory: Factory creating LCPEncryptionConfiguration instance
+        :type configuration_factory: api.config.ConfigurationFactory
+
+        :param hasher_factory: Factory responsible for creating Hasher implementations
+        :type hasher_factory: hash.HasherFactory
+        """
+        self._configuration_storage = configuration_storage
+        self._configuration_factory = configuration_factory
+        self._hasher_factory = hasher_factory
+        self._hasher_instance = None
+
+    def _get_hasher(self, configuration):
+        """Returns a Hasher instance
+
+        :param configuration: Configuration object
+        :type configuration: LCPServerConfiguration
+
+        :return: Hasher instance
+        :rtype: hash.Hasher
+        """
+        if self._hasher_instance is None:
+            self._hasher_instance = self._hasher_factory.create(
+                configuration.encryption_algorithm)
+
+        return self._hasher_instance
+
+    def _create_partial_license(self, configuration, patron_id=None, license_start=None, license_end=None):
+        """Creates a partial LCP license used an input by the LCP License Server for generation of LCP licenses
+
+        :param configuration: Configuration object
+        :type configuration: LCPServerConfiguration
+
+        :param patron_id: Patron's unique ID
+        :type patron_id: Optional[string]
+
+        :param license_start: Date and time when the license begins
+        :type license_start: Optional[datetime.datetime]
+
+        :param license_end: Date and time when the license ends
+        :type license_end: Optional[datetime.datetime]
+
+        :return: Partial LCP license
+        :rtype: Dict
+        """
+        hasher = self._get_hasher(configuration)
+        encrypted_password = hasher.hash(self.DEFAULT_PATRON_PASSPHRASE)
+        partial_license = {
+            'provider': configuration.provider_name,
+            'encryption': {
+                'user_key': {
+                    'text_hint': configuration.passphrase_hint,
+                    'hex_value': encrypted_password,
+                }
+            },
+            'rights': {
+                'print': int(configuration.max_printable_pages),
+                'copy': int(configuration.max_copiable_pages)
+            }
+        }
+
+        if patron_id:
+            partial_license['user'] = {
+                'id': str(patron_id)
+            }
```

Comment:
I don't want to expose our internal database IDs to an external service. It would be more appropriate to create a permanent `Credential` object for a given `Patron` and `Collection`, and pass `Credential.credential` to the LCP server. The `type` of the collection could be "User identiifer passed to LCP server".

See `patron_remote_identifier` in api/rbdigital.py for a similar example: we create an alias for the patron to use when we talk about them to the RBDigital API.

Reply:
Add `Credential` generation using `uuid` module